tox.ini

```diff
@@ -0,0 +1,111 @@
+  setuptools >= 41.4.0
+  pip >= 19.3.0
+skip_missing_interpreters = True
+# `usedevelop = true` overrides `skip_install` instruction, it's unwanted
+usedevelop = false
+
+[testenv]
+description =
+  Run the tests under {basepython} and
+  devel: ansible devel branch
+  ansible29: ansible 2.9
+  core: ansible-base 2.10
+  py: ansible-core 2.11
+extras =
+  test
+deps =
+  ansible29: ansible>=2.9,<2.10
+  py: ansible-core>=2.11
+  devel: ansible-core @ git+https://github.com/ansible/ansible.git  # GPLv3+
+commands =
+  # safety measure to assure we do not accidentally run tests with broken dependencies
+  {envpython} -m pip check
+  # We add coverage options but not making them mandatory as we do not want to force
+  # pytest users to run coverage when they just want to run a single test with `pytest -k test`
+  {envpython} -m pytest \
+  --junitxml "{toxworkdir}/junit.{envname}.xml" \
+  {posargs:\
+    --cov ansiblecompat \
+    --cov "{envsitepackagesdir}/ansiblecompat" \
+    --cov-report term-missing:skip-covered \
+    --cov-report xml:.test-results/pytest/cov.xml \
+    --no-cov-on-fail}
+passenv =
+  CURL_CA_BUNDLE  # https proxies, https://github.com/tox-dev/tox/issues/1437
+  FORCE_COLOR
+  HOME
+  NO_COLOR
+  PYTEST_*  # allows developer to define their own preferences
+  PY_COLORS
+  REQUESTS_CA_BUNDLE  # https proxies
+  SSL_CERT_FILE  # https proxies
+  LANG
+  LC_ALL
+  LC_CTYPE
+# recreate = True
+setenv =
+  COVERAGE_FILE = {env:COVERAGE_FILE:{toxworkdir}/.coverage.{envname}}
+  PIP_DISABLE_PIP_VERSION_CHECK = 1
+  PIP_CONSTRAINT = {toxinidir}/constraints.txt
+  PRE_COMMIT_COLOR = always
+  FORCE_COLOR = 1
+allowlist_externals =
+  sh
+
+[testenv:lint]
+description = Run all linters
+basepython = python3
+deps =
+  pre-commit>=2.6.0
+  pip-tools>=5.4.0
+  setuptools>=51.1.1
+skip_install = true
+commands =
+  pip-compile --extra test --output-file=constraints.txt setup.py
+  {envpython} -m pre_commit run -a
+passenv =
+  {[testenv]passenv}
+  PRE_COMMIT_HOME
+
+[testenv:deps]
+description = Bump all test depeendencies
+basepython = python3
+deps =
+  pre-commit>=2.6.0
+  pip-tools>=5.4.0
+  setuptools>=51.1.1
+skip_install = true
+commands =
+  pip-compile --upgrade --extra test --output-file=constraints.txt setup.py
+  {envpython} -m pre_commit run {posargs:--all-files --hook-stage manual -v}
+
+[testenv:packaging]
+basepython = python3
+description =
+  Build package, verify metadata, install package and assert behavior when ansible is missing.
+deps =
+  pep517 >= 0.7.0
+  twine
+skip_install = true
+# Ref: https://twitter.com/di_codes/status/1044358639081975813
+commands =
+  # build wheel and sdist using PEP-517
+  {envpython} -c 'import os.path, shutil, sys; \
+  dist_dir = os.path.join("{toxinidir}", "dist"); \
+  os.path.isdir(dist_dir) or sys.exit(0); \
+  print("Removing \{!s\} contents...".format(dist_dir), file=sys.stderr); \
+  shutil.rmtree(dist_dir)'
+  {envpython} -m pep517.build \
+    --source \
+    --binary \
```

Comment:
When you migrate to `build`, drop these because starting with v0.5.0, it has a special logic to build wheels from sdists which is preferable for testing reproducibility.